File: app/src/main/java/ar/com/develup/tateti/actividades/ActividadPartidas.java
package ar.com.develup.tateti.actividades;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;

import com.google.firebase.database.ChildEventListener;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.GenericTypeIndicator;

import ar.com.develup.tateti.R;
import ar.com.develup.tateti.adaptadores.AdaptadorPartidas;
import ar.com.develup.tateti.modelo.Constantes;
import ar.com.develup.tateti.modelo.Partida;
import butterknife.BindView;
import butterknife.OnClick;

/**
 * Created by maribelmai on 26/3/17.
 */

public class ActividadPartidas extends ActividadBasica {

    private static final String TAG = ActividadPartidas.class.getSimpleName();
    @BindView(R.id.partidas)
    RecyclerView partidasRecycler;

    private AdaptadorPartidas adaptadorPartidas;
    private ChildEventListener listenerTablaPartidas = new ChildEventListener() {

        @Override
        public void onChildAdded(DataSnapshot dataSnapshot, String s) {
            Log.i(TAG, "onChildAdded: " + dataSnapshot);

            GenericTypeIndicator<Partida> typeIndicator = new GenericTypeIndicator<Partida>() {};
            Partida partida = dataSnapshot.getValue(typeIndicator);
            partida.setId(dataSnapshot.getKey());
            adaptadorPartidas.agregarPartida(partida);
        }

        @Override
        public void onChildChanged(DataSnapshot dataSnapshot, String s) {
            Log.i(TAG, "onChildChanged: " + s);
        }

        @Override
        public void onChildRemoved(DataSnapshot dataSnapshot) {
            Log.i(TAG, "onChildRemoved: ");
        }

        @Override
        public void onChildMoved(DataSnapshot dataSnapshot, String s) {
            Log.i(TAG, "onChildMoved: " + s);
        }

        @Override
        public void onCancelled(DatabaseError databaseError) {
            Log.i(TAG, "onCancelled: ");
        }
    };

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.adaptadorPartidas = new AdaptadorPartidas(this);
        this.partidasRecycler.setLayoutManager(new LinearLayoutManager(this));
        this.partidasRecycler.setAdapter(adaptadorPartidas);
    }

    @Override
    protected void onResume() {
        super.onResume();
        FirebaseDatabase.getInstance().getReference().child(Constantes.TABLA_PARTIDAS).addChildEventListener(this.listenerTablaPartidas);
    }

    @Override
    protected void onPause() {
        super.onPause();
        FirebaseDatabase.getInstance().getReference().child(Constantes.TABLA_PARTIDAS).removeEventListener(this.listenerTablaPartidas);
    }

    @Override
    protected int getLayout() {
        return R.layout.actividad_partidas;
    }

    @OnClick(R.id.nuevaPartida)
    public void nuevaPartida() {
        Intent intent = new Intent(this, ActividadPartida.class);
        startActivity(intent);
    }
}

File: app/src/main/java/ar/com/develup/tateti/actividades/ActividadPartidas.kt
package ar.com.develup.tateti.actividades

import android.content.Intent
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import ar.com.develup.tateti.R
import ar.com.develup.tateti.adaptadores.AdaptadorPartidas
import ar.com.develup.tateti.modelo.Constantes
import ar.com.develup.tateti.modelo.Partida
import com.google.firebase.database.ChildEventListener
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.FirebaseDatabase
import kotlinx.android.synthetic.main.actividad_partidas.*

class ActividadPartidas : AppCompatActivity() {

    private lateinit var adaptadorPartidas: AdaptadorPartidas

    private val listenerTablaPartidas: ChildEventListener = object : ChildEventListener {
        override fun onChildAdded(dataSnapshot: DataSnapshot, s: String?) {
            Log.i(TAG, "onChildAdded: $dataSnapshot")
            val partida = dataSnapshot.getValue(Partida::class.java)!!
            partida.id = dataSnapshot.key
            adaptadorPartidas.agregarPartida(partida)
        }

        override fun onChildChanged(dataSnapshot: DataSnapshot, s: String?) {
            Log.i(TAG, "onChildChanged: $s")
            val partida = dataSnapshot.getValue(Partida::class.java)!!
            partida.id = dataSnapshot.key
            adaptadorPartidas.partidaCambio(partida)
        }

        override fun onChildRemoved(dataSnapshot: DataSnapshot) {
            Log.i(TAG, "onChildRemoved: ")
            val partida = dataSnapshot.getValue(Partida::class.java)!!
            partida.id = dataSnapshot.key
            adaptadorPartidas.remover(partida)
        }

        override fun onChildMoved(dataSnapshot: DataSnapshot, s: String?) {
            Log.i(TAG, "onChildMoved: $s")
        }

        override fun onCancelled(databaseError: DatabaseError) {
            Log.i(TAG, "onCancelled: ")
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.actividad_partidas)
        adaptadorPartidas = AdaptadorPartidas(this)
        partidas.layoutManager = LinearLayoutManager(this)
        partidas.adapter = adaptadorPartidas
        nuevaPartida.setOnClickListener { nuevaPartida() }
    }

    override fun onResume() {
        super.onResume()
        FirebaseDatabase.getInstance().reference.child(Constantes.TABLA_PARTIDAS).addChildEventListener(listenerTablaPartidas)
    }

    fun nuevaPartida() {
        val intent = Intent(this, ActividadPartida::class.java)
        startActivity(intent)
    }

    companion object {
        private val TAG = ActividadPartidas::class.java.simpleName
    }
}